docs/rfcs/XXX-pull-request-review.md

```diff
@@ -0,0 +1,145 @@
+# Feature title
+
+Pull Request Review
+
+## Status
+
+Proposed
+
+## Summary
+
+Give and receive code reviews on pull requests within Atom.
+
+## Motivation
+
+Workflows around pull request reviews involve many trips between your editor and your browser. If you check out a pull request locally to test it and want to leave comments, you need to map the issues that you've found in your working copy back to lines on the diff to comment appropriately. Similarly, when you're given a review, you have to mentally correlate review comments on the diff on GitHub with the corresponding lines in your local working copy, then map _back_ to diff lines to respond once you've established context. By revealing review comments as decorations directly within the editor, we can eliminate all of these round-trips and streamline the review process for all involved.
+
+Peer review is also a critical part of the path to acceptance for pull requests in many common workflows. By surfacing progress through code review, we provide context on the progress of each unit of work alongside existing indicators like commit status.
+
+## Explanation
+
+### Current pull request tile
+
+Reviews on the current pull request are rendered as a list on the current pull request tile.
+
+![review-list](https://user-images.githubusercontent.com/378023/46273505-b9533280-c590-11e8-840e-a8eac8023cad.png)
+
+* The review summary bubble is elided after the first sentence or N characters if necessary.
+* Clicking the review summary bubble opens an `IssueishPaneItem` in the workspace center, open to the reviews tab.
+* Clicking a line comment opens or activates an editor on the referenced file and scrolls to center the comment's line, translated according to local changes if appropriate.
+* Line comments within the review are rendered: _with a dot_ before the file has been opened and the corresponding decoration is visible; _with no icon_ after the file and decoration have been seen; and _with a checkmark_ after the comment has been marked "resolved" with the control on its decoration.
+
+If a new review has been started locally, it appears at the top of the "Reviews" section within this tile:
+
+![pending-review](https://user-images.githubusercontent.com/378023/46275946-9bd69680-c599-11e8-9889-66c35458286a.png)
+
+* The review summary is a TextEditor that may be used to compose a summary comment.
+* Choosing "Cancel" dismisses the review and any comments made. If there are local review comments that will be lost, a confirmation prompt is shown first.
+* Choosing "Submit review" submits the drafted review to GitHub.
+
+### Non-current pull request tiles
+
+Pull request tiles other than the current pull request display a one-line review summary, showing the number of accepting, comment, and change-request reviews made on each. Clicking the review summary opens the `IssueishPaneItem` for that pull request and opens the reviews tab.
```

Comment:
Is the main purpose to have a quick way to navigate to the "Reviews" tab? Or to keep an eye on new reviews? For the later, some sort of "new comments since I last visited", might be handy, but comes with its own challenges. It also looks a bit crowded with all the icons and numbers.

Reply:
> Is the main purpose to have a quick way to navigate to the "Reviews" tab? Or to keep an eye on new reviews?

Yes 😆 It's also to start using the PR result tiles to show a bit more information without having to click through, and to give a little symmetry to the information you see for current and non-current PRs.

I'm not 100% sold on it myself though mostly because:

> It also looks a bit crowded with all the icons and numbers.

So if it's just too messy we can yank it.

Sidenote:

> For the later, some sort of "new comments since I last visited", might be handy, but comes with its own challenges.

I'm deliberately punting anything related to notifications about new information arriving from this RFC. It's still important and a place we can add a ton of value, but I believe it's something we should address in a consistent way across the package as a whole (controls that let you opt-in to Atom notifications? Blue-bars to signal "there's something new here"?). It's also something that's going to take a lot of us getting it wrong before we figure out how to get it right.

Reply:
> It's also to start using the PR result tiles to show a bit more information without having to click through

Maybe we could show the "reviews" progress bar?

![image](https://user-images.githubusercontent.com/378023/46338283-af9dfd80-c66a-11e8-917a-0e0972c7ca43.png)

And maaaaybeeee also a "tasks" bar.

<img width="362" alt="non-current pull request tile" src="https://user-images.githubusercontent.com/378023/46338340-e247f600-c66a-11e8-98c1-850f58edbaa9.png">

Then you can glance over the PRs and get a sense of their progress. Like " 😱 Oh, this looks almost ready, I better add my review."

Reply:
Okay, I definitely like yours better than my mockup, but I'm still not convinced we shouldn't just leave non-current PR tiles alone. I'm kind of leaning toward not touching it I guess?